Client/Frontend/Home/HomeTabBanner.swift

```diff
@@ -0,0 +1,240 @@
+            
+            image.topAnchor.constraint(equalTo: cardView.topAnchor, constant: 48),
+            image.leadingAnchor.constraint(equalTo: cardView.leadingAnchor, constant: 16),
+            image.widthAnchor.constraint(equalToConstant: UX.logoSize.width),
+            image.heightAnchor.constraint(equalToConstant: UX.logoSize.height),
+
+            bannerTitle.topAnchor.constraint(equalTo: image.topAnchor, constant: -16),
+            bannerTitle.leadingAnchor.constraint(equalTo: image.trailingAnchor, constant: 16),
+            bannerTitle.trailingAnchor.constraint(equalTo: cardView.trailingAnchor),
+
+            descriptionText.topAnchor.constraint(equalTo: bannerTitle.bottomAnchor),
+            descriptionText.leadingAnchor.constraint(equalTo: bannerTitle.leadingAnchor),
+            descriptionText.trailingAnchor.constraint(equalTo: cardView.trailingAnchor, constant: -12),
+            descriptionText.bottomAnchor.constraint(greaterThanOrEqualTo: CTAButton.topAnchor, constant: -16),
+
+            dismissButton.topAnchor.constraint(equalTo: cardView.topAnchor, constant: 16),
+            dismissButton.trailingAnchor.constraint(equalTo: cardView.trailingAnchor, constant: -16),
+            dismissButton.heightAnchor.constraint(equalToConstant: 16),
+            dismissButton.widthAnchor.constraint(equalToConstant: 16),
+
+            CTAButton.centerXAnchor.constraint(equalTo: cardView.centerXAnchor),
+            CTAButton.bottomAnchor.constraint(equalTo: cardView.bottomAnchor, constant: -16),
+            CTAButton.widthAnchor.constraint(equalToConstant: UX.learnHowButtonSize.width),
+            CTAButton.heightAnchor.constraint(equalToConstant: UX.learnHowButtonSize.height)
+        ])
+    }
+    
+    /// Apply message data, including handling of cases where certain parts of the message are missing.
+    private func applyMessage() {
+        
+        /// If no messages exist, continue using our evergreen message.
+        guard let message = message else {
+            bannerTitle.text = String.DefaultBrowserCardTitle
+            descriptionText.text = String.DefaultBrowserCardDescription
+            CTAButton.setTitle(String.DefaultBrowserCardButton, for: .normal)
+            
+            return
+        }
+        
+        guard let buttonLabel = message.data.buttonLabel else {
+            dismissClosure?()
+            
+            messagingManager.onMalformedMessage(messageKey: message.Id)
+            
+            return
+        }
+        
+        guard let title = message.data.title else {
+            cardView.willRemoveSubview(bannerTitle)
+            bannerTitle.removeFromSuperview()
+            
+            descriptionText.text = message.data.text
+            CTAButton.setTitle(buttonLabel, for: .normal)
+            
+            messagingManager.onMessageDisplayed(message)
+            
+            return
+        }
```

Comment:
I think using `guard`s here makes things harder to reason about. Use an `if` and `if let` statements to handle each part.

```swift
if let buttonLabel = message.data.buttonLabel {
    CTAButton.setTitle(buttonLabel, for: .normal)
} else {
    cardView.willRemoveSubview(bannerTitle)
    bannerTitle.removeFromSuperview()
    cardView.addTarget(self, action: #selector(self.handleCTA), for: .touchUpInside)
}

if let title = message.data.title {
    bannerTitle.text = text
} else {
   cardView.willRemoveSubview(bannerTitle)
   bannerTitle.removeFromSuperView()
}

descriptionText.text = message.data.text
messagingManager.onMessageDisplayed(message)
```